packages/esm-service-queues-app/src/send-patient-back-toqueue/send-back-patient-toqueue.component.tsx

```diff
@@ -0,0 +1,288 @@
+  const defaultResultReviewUrgencyStatus = concepts?.defaultResultReviewPriorityConceptUuid;
+
+  const schema = useMemo(
+    () =>
+      z.object({
+        location: z.string({ required_error: t('queueLocationRequired', 'Queue location is required') }),
+        service: z.string({ required_error: t('serviceIsRequired', 'Service is required') }),
+        status: z.string({ required_error: t('statusIsRequired', 'Status is required') }),
+        priority: z.string({ required_error: t('priorityIsRequired', 'Priority is required') }),
+      }),
+    [t],
+  );
+
+  type ChangeStatusForm = z.infer<typeof schema>;
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+    getValues,
+  } = useForm<ChangeStatusForm>({
+    resolver: zodResolver(schema),
+  });
+
+  const queueServiceLocation = data?.locationWaitingFor?.display ?? '';
+  const { queues } = useQueues(queueServiceLocation);
+  const { queueLocations } = useQueueLocations();
+  const { mutateQueueEntries } = useMutateQueueEntries();
+
+  const onSubmit = async (updateQueueData: ChangeStatusForm) => {
+    const { status, service } = updateQueueData;
+    const defaultPriority = concepts.defaultPriorityConceptUuid;
+    const queuePriority = updateQueueData.priority === '' ? defaultPriority : updateQueueData.priority;
+    const emergencyPriorityConceptUuid = concepts.emergencyPriorityConceptUuid;
+    const sortWeight = updateQueueData.priority === emergencyPriorityConceptUuid ? 1.0 : 0.0;
+    const endDate = new Date();
+
+    try {
+      const response = await updateQueueEntry(
+        data?.visit.uuid,
+        data?.queue.uuid,
+        service,
+        data?.uuid,
+        patientUuid,
+        queuePriority,
+        status,
+        endDate,
+        sortWeight,
+      );
+
+      if (response.status === 201) {
+        showSnackbar({
+          isLowContrast: true,
+          title: t('sendPatientBack', 'Back to queue'),
+          kind: 'success',
+          subtitle: t('sendBackPatientBackSuccess', 'Patient sent back to queue Successfully'),
+        });
+        closeModal();
+        mutateQueueEntries();
+        navigate({ to: `${window.spaBase}/home/service-queues` });
+      }
+    } catch (error) {
+      showSnackbar({
+        title: t('queueEntryStatusUpdateFailed', 'Error updating queue entry status'),
+        kind: 'error',
+        subtitle: error?.message,
+      });
+    }
+  };
+
+  const onError = (errors: any) => console.error(errors);
+
+  if (!data) {
+    return (
+      <Tile>
+        <ModalHeader closeModal={closeModal} title={t('sendPatientToAQueue', 'Add to a queue')} />
+        <ModalBody>
+          <div className={styles.modalBody}>
+            <h5>{t('patientNotInQueue', 'The patient is not in the queue')}</h5>
+          </div>
+        </ModalBody>
+      </Tile>
+    );
+  }
+
+  return (
+    <div>
+      <Form onSubmit={handleSubmit(onSubmit, onError)}>
```

Comment:
It'd be nice if we can rewrite our already existing forms to add / transition queue entries so we are just mostly reusing that code.

Reply:
+1

Reply:
I can make the form for transition and for this component to use a generic form which we can pass the props

Reply:
@chibongho 

![change](https://github.com/user-attachments/assets/989ee9fa-f2dd-4dd9-bbac-d9797e6b995d)

Reply:
I think i have understood what is required let me refactor everything and use the `QueueEntryActionModal` @ojwanganto @chibongho

Reply:
![updated](https://github.com/user-attachments/assets/00172b67-1839-4d5b-b631-678e05e7ffe1)


@ojwanganto @chibongho I have reused the component and move the add patient button to queue